src/interfaces.ts

```diff
@@ -318,3 +276,260 @@ export interface ResolvedModule {
+  ): any;
+
+  /**
+   * Resolve a request like a require expression.
+   */
+  resolveSync(context: string, request: string): string;
+
+  /**
+   * Adds a file as dependency of the loader result in order to make them watchable.
+   * For example, html-loader uses this technique as it finds src and src-set attributes.
+   * Then, it sets the url's for those attributes as dependencies of the html file that is parsed.
+   */
+  addDependency(file: string): void;
+
+  /**
+   * Adds a file as dependency of the loader result in order to make them watchable.
+   * For example, html-loader uses this technique as it finds src and src-set attributes.
+   * Then, it sets the url's for those attributes as dependencies of the html file that is parsed.
+   */
+  dependency(file: string): void;
+
+  /**
+   * Add a directory as dependency of the loader result.
+   */
+  addContextDependency(directory: string): void;
+
+  /**
+   * Remove all dependencies of the loader result. Even initial dependencies and these of other loaders. Consider using pitch.
+   */
+  clearDependencies(): void;
+
+  /**
+   * Pass values to the next loader.
+   * If you know what your result exports if executed as module, set this value here (as a only element array).
+   */
+  value: any;
+
+  /**
+   * Passed from the last loader.
+   * If you would execute the input argument as module, consider reading this variable for a shortcut (for performance).
+   */
+  inputValue: any;
+
+  /**
+   * A boolean flag. It is set when in debug mode.
+   */
+  debug: boolean;
+
+  /**
+   * Should the result be minimized.
+   */
+  minimize: boolean;
+
+  /**
+   * Should a SourceMap be generated.
+   */
+  sourceMap: boolean;
+
+  /**
+   * Target of compilation. Passed from configuration options.
+   * Example values: "web", "node"
+   */
+  target:
+    | 'web'
+    | 'webworker'
+    | 'async-node'
+    | 'node'
+    | 'electron-main'
+    | 'electron-renderer'
+    | 'node-webkit'
+    | string;
+
+  /**
+   * This boolean is set to true when this is compiled by webpack.
+   *
+   * Loaders were originally designed to also work as Babel transforms.
+   * Therefore if you write a loader that works for both, you can use this property to know if
+   * there is access to additional loaderContext and webpack features.
+   */
+  webpack: boolean;
+
+  /**
+   * Emit a file. This is webpack-specific.
+   */
+  emitFile(name: string, content: Buffer | string, sourceMap: any): void;
+
+  /**
+   * Access to the compilation's inputFileSystem property.
+   */
+  fs: any;
+
+  /**
+   * Which mode is webpack running.
+   */
+  mode: 'production' | 'development' | 'none';
+
+  /**
+   * Hacky access to the Compilation object of webpack.
+   */
+  _compilation: any;
+
+  /**
+   * Hacky access to the Compiler object of webpack.
+   */
+  _compiler: any;
+
+  /**
+   * Hacky access to the Module object being loaded.
+   */
+  _module: any;
+
+  /** Flag if HMR is enabled */
+  hot: boolean;
+}
```

Comment:
I think better to avoid using this, because it is invalid interface, just keep copy only necessary properties, I think we need improve this on webpack side